addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/ScheduleHelper.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.evohome.handler;
+
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.openhab.binding.evohome.internal.api.models.v2.response.DailySchedule;
+import org.openhab.binding.evohome.internal.api.models.v2.response.DailySchedules;
+import org.openhab.binding.evohome.internal.api.models.v2.response.SwitchPoint;
+
+/**
+ * Class to find current and future schedules.
+ *
+ * @author Jeroen Peters
+ *
+ */
+public class ScheduleHelper {
+
+    private DailySchedules schedules;
+
+    public ScheduleHelper(DailySchedules schedules) {
+        this.schedules = schedules;
+    }
+
+    public boolean hasSchedule() {
+        boolean result = false;
+
+        for (DailySchedule schedule : schedules) {
+            if (schedule.getSwitchPoints().size() > 0) {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    public ZonedDateTime getStartNextSwitchPoint(ZonedDateTime zoned) {
+        // Map zoned day index to schedule list (Monday = 0, Sunday = 6)
+        int todayIndex = zoned.getDayOfWeek().getValue() - 1;
+        LocalTime time = zoned.toLocalTime();
+
+        List<SwitchPoint> switchPoints = schedules.get(todayIndex).getSwitchPoints();
+        for (SwitchPoint switchPoint : switchPoints) {
+            if (time.compareTo(switchPoint.getTimeOfDay()) < 0) {
+                return zoned.with(switchPoint.getTimeOfDay());
```

Comment:
Personally, I'm not too fond of multiple returns in one method/fuction. Rather set a return variable and return at the end of the method.